### PR TITLE
Added an abstract field to the idea headers

### PR DESCRIPTION
**Pull request by YashdalfTheGray**
@bcbrennecke, you shouldn't be blocked anymore. 

Edit: it takes the first 12 words from the description of the idea and passes it with the idea headers in a brand new, shinny field called `abstract` (ooohhhh, aaaaahhh). 

Also added
 
* A note about `generate:data` that was missing from the `gulp usage` task
* Changed the indentation to 4 spaces for `ideas.js`

Requested reviewers: @Mctalian, @bcbrennecke

### CODE REVIEW

File: server/ideas.js [hunk 1]
```diff
@@ -43,131 +44,133 @@ exports.create = function(id, title, description, author, likes, comments, backs
...
+            var headers = [];
+            for(var i = 0; i < docs.length; i++) {
+                var descFirstWords = _.take(_.words(docs[i].description), 12);
+
+                headers.push({
+                    id: docs[i].ideaId,
+                    title: docs[i].title,
+                    author: docs[i].author,
+                    likes: docs[i].likes.length,
+                    abstract: descFirstWords.join(' ')
```
Comment (YashdalfTheGray) [inline server/ideas.js]: This line and line 139 is where the actual changes are.

Comment (Mctalian) [inline server/ideas.js]: What does it do? Takes the first 12 words?

Comment (YashdalfTheGray) [inline server/ideas.js]: Yup!

**Comment by Mctalian** [comment]
Could want more words in the future, but a great start.

Approved.

**Comment by bcbrennecke** [comment]
I'd love to add a couple more words than 12 to give users a little bit more of an idea.  I'd vote for 30.  But yes in my expert code reviewing opinion, this looks good.  I'll get it added to the cards page once it's merged.

**Comment by Mctalian** [comment]
I agree that something around 30 might be more helpful.

**Comment by YashdalfTheGray** [comment]
I can do 30.

**Comment by YashdalfTheGray** [comment]
Changed.

**Pull request merged**